src/transformers/models/grounding_dino/processing_grounding_dino.py

```diff
@@ -149,3 +175,21 @@ def model_input_names(self):
         tokenizer_input_names = self.tokenizer.model_input_names
         image_processor_input_names = self.image_processor.model_input_names
         return list(dict.fromkeys(tokenizer_input_names + image_processor_input_names))
+    
+    def post_process_grounded_object_detection(self, outputs, input_ids, box_threshold: float, text_threshold: float, target_sizes: Union[TensorType, List[Tuple]] = None):
```

Comment:
Thanks for adding, one remark here is that currently the result is not consistent with `post_process_object_detection`. The latter gives me the following:
```
[{'score': 0.4526309669017792, 'label': 'a cat.', 'box': {'xmin': 344, 'ymin': 23, 'xmax': 637, 'ymax': 373}}, {'score': 0.40809395909309387, 'label': 'a cat.', 'box': {'xmin': 11, 'ymin': 51, 'xmax': 316, 'ymax': 472}}]
```
i.e. a list of dictionaries, each containing a score, label and bounding box.

However this method gives me this:
```
{'scores': tensor([0.4526, 0.4081]), 'labels': ['a cat', 'a cat'], 'boxes': tensor([[344.8151,  23.1787, 637.3994, 373.8306], [ 11.9177,  51.5846, 316.5746, 472.8922]])}
```
=> would be great to have the same API (and preferably use the first option)

Reply:
@NielsRogge `GroundingDINOImageProcessor.post_process_object_detection`, which is the same as `Owl-ViT`, outputs for that cat image and text as "a cat.":

```python
[{'scores': tensor([0.4526, 0.4081]), 'labels': tensor([1, 1]), 'boxes': tensor([[0.5388, 0.0483, 0.9959, 0.7788],
        [0.0186, 0.1075, 0.4946, 0.9852]])}]
```

and `GroundingDINOProcessor.post_process_grounded_object_detection` for the same input returns:

```python
[{'scores': tensor([0.4526, 0.4081]), 'labels': ['a cat', 'a cat'], 'boxes': tensor([[0.5388, 0.0483, 0.9959, 0.7788],
        [0.0186, 0.1075, 0.4946, 0.9852]])}]
```

So both are a list of dict with length `batch_size` where the values in the dict are tensors with `num_boxes` (i.e. the amount of boxes detected for the given box threshold)

Isn't that the expected behavior?

Code to reproduce the values I mentioned above:

```python
>>> import torch
>>> from transformers import GroundingDINOForObjectDetection, GroundingDINOProcessor
>>> from PIL import Image
>>> import requests
>>> checkpoint = "EduardoPacheco/grounding-dino-tiny"
>>> model = GroundingDINOForObjectDetection.from_pretrained(checkpoint)
>>> processor = GroundingDINOProcessor.from_pretrained(checkpoint)
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
>>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
>>> image = Image.open(requests.get(url, stream=True).raw).convert("RGB")
>>> text = "a cat."
>>> inputs = processor(text=text, images=image, return_tensors="pt")
>>> with torch.no_grad():
...     outputs = model(**inputs)
... 
>>> processor.image_processor.post_process_object_detection(outputs, 0.35)
[{'scores': tensor([0.4526, 0.4081]), 'labels': tensor([1, 1]), 'boxes': tensor([[0.5388, 0.0483, 0.9959, 0.7788],
        [0.0186, 0.1075, 0.4946, 0.9852]])}]
>>> 
>>> 
>>> processor.post_process_grounded_object_detection(outputs, inputs.input_ids, 0.35, 0.3)
[{'scores': tensor([0.4526, 0.4081]), 'labels': ['a cat', 'a cat'], 'boxes': tensor([[0.5388, 0.0483, 0.9959, 0.7788],
        [0.0186, 0.1075, 0.4946, 0.9852]])}]
```

Reply:
Apologies, I was looking at the outputs of the object detection pipeline in my notebook. Ok so it's fine, awesome!

Could you add a corresponding test for it in tests/test_processor_grounding_dino.py?

Reply:
@NielsRogge Added tests for `GroundingDINOImageProcessor` and `GroundingDINOProcessor` and added the `post_process_grounded_object_detection` to the `test_inference_object_detection_head` test at `tests/models/grounding_dino/test_modeling_grounding_dino.py`

Reply:
Thanks, feel free to close this comment